crates/core/component/fee/src/component/view.rs

```diff
@@ -20,11 +16,24 @@ pub trait StateReadExt: StateRead {
             .ok_or_else(|| anyhow!("Missing FeeParameters"))
     }
 
-    /// Gets the gas prices from the JMT.
+    /// Gets the current gas prices for the fee token.
     async fn get_gas_prices(&self) -> Result<GasPrices> {
-        self.get(state_key::gas_prices())
-            .await?
-            .ok_or_else(|| anyhow!("Missing GasPrices"))
+        // When we implement dynamic gas pricing, we will want
+        // to read the prices we computed. But until then, we need to
+        // read these from the _fee params_ instead, since those are
+        // the values that will get updated by governance.
+        let params = self.get_fee_params().await?;
+        Ok(params.fixed_gas_prices)
+    }
+
+    /// Gets the current gas prices for alternative fee tokens.
+    async fn get_alt_gas_prices(&self) -> Result<Vec<GasPrices>> {
+        // When we implement dynamic gas pricing, we will want
+        // to read the prices we computed. But until then, we need to
+        // read these from the _fee params_ instead, since those are
+        // the values that will get updated by governance.
+        let params = self.get_fee_params().await?;
+        Ok(params.fixed_alt_gas_prices)
```

Comment:
This fixes a critical bug in the original fee component logic, which would prevent fees from ever being changed post-genesis.